src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -1,33 +1,114 @@
+            if (isNotTimeOrDateFormat) {
+                throw new ParseException(
+                        String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.INVALID_DATE_OR_TIME_MESSAGE));
+            }
+        }
+        return new FindCommand(new DeadlineContainsKeywordsPredicate(Arrays.asList(keywords)));
+    }
+
+    private static FindCommand findByPriority(String[] keywords) throws ParseException {
+        for (String keyword : keywords) {
+            ParserUtil.parsePriority(keyword);
+        }
+        return new FindCommand(new PriorityContainsKeywordsPredicate(Arrays.asList(keywords)));
+    }
+
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        String[] keywords;
         String trimmedArgs = args.trim();
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DEADLINE, PREFIX_MODULE_CODE, PREFIX_PRIORITY);
+
+        boolean isPreambleMissing = argMultimap.getPreamble().isEmpty();
+        boolean isPrefixNamePresent = argMultimap.getValue(PREFIX_NAME).isPresent();
+        boolean isPrefixDeadlinePresent = argMultimap.getValue(PREFIX_DEADLINE).isPresent();
+        boolean isPrefixModuleCodePresent = argMultimap.getValue(PREFIX_MODULE_CODE).isPresent();
+        boolean isPrefixPriorityPresent = argMultimap.getValue(PREFIX_PRIORITY).isPresent();
+
         if (trimmedArgs.isEmpty()) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        } else if (moreThanOnePrefixPresent(
+                argMultimap, PREFIX_NAME, PREFIX_MODULE_CODE, PREFIX_DEADLINE, PREFIX_PRIORITY)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MORE_THAN_ONE_PREFIX_MESSAGE));
+        } else if (isPrefixNamePresent && isPreambleMissing) {
+            keywords = argMultimap.getValue(PREFIX_NAME).get().split("\\s+");
+            return findByName(keywords);
+        } else if (isPrefixModuleCodePresent && isPreambleMissing) {
+            keywords = argMultimap.getValue(PREFIX_MODULE_CODE).get().split("\\s+");
+            return findByModuleCode(keywords);
+        } else if (isPrefixDeadlinePresent && isPreambleMissing) {
+            keywords = argMultimap.getValue(PREFIX_DEADLINE).get()
+                    .split("\\s+");
+            return findByDeadline(keywords);
+        } else if (isPrefixPriorityPresent && isPreambleMissing) {
+            keywords = argMultimap.getValue(PREFIX_PRIORITY).get().split("\\s+");
+            return findByPriority(keywords);
```

Comment:
The fetching of keywords for each field appears to have some overlaps. You could consider abstracting the implementation and creating a method called getKeywords(String prefix) to improve the code quality :)

Reply:
Amended!